crates/proto/src/domain/account.rs

```diff
@@ -440,12 +517,17 @@ impl AccountStorageMapDetails {
         }
     }
 
-    fn from_specific_keys(slot_index: u8, keys: &[Word], storage_map: &StorageMap) -> Self {
-        if keys.len() > Self::MAX_RETURN_ENTRIES {
-            Self::too_many_entries(slot_index)
-        } else {
-            // TODO For now, we return all entries instead of specific keys with proofs
-            Self::from_all_entries(slot_index, storage_map)
+    /// Creates an `AccountStorageMapDetails` from already-queried entries (e.g., from database).
+    /// This is useful when entries have been fetched directly rather than extracted from a
+    /// `StorageMap`.
+    pub fn from_entries(slot_index: u8, map_entries: Vec<(Word, Word)>) -> Self {
+        let too_many_entries = map_entries.len() > Self::MAX_RETURN_ENTRIES;
+        let map_entries = if too_many_entries { Vec::new() } else { map_entries };
```

Comment:
Similar to one of the above comments: we should probably optimize this in the future to avoid reading full maps from the database when there are too many entries - but that would be even more tricky than doing this for the asset vault.